bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -420,13 +412,20 @@ private Control createDialog(final Composite parent) {
 		createSearchBar();
 		createSearchTools();
 		createCloseTools();
+		initializeSearchShortcutHandlers();
 
 		container.layout();
 
 		applyDialogFont(container);
 		return container;
 	}
 
+	private void initializeSearchShortcutHandlers() {
```

Comment:
I would expect this method to be close in the code to this method https://github.com/eclipse-platform/eclipse.platform.ui/pull/2078/files#diff-05a02220b3b1c63bb5dac5a281641fc7a61fc1bb0d83ddfa3d65d0608f7b1c4aR695, increasing code cohesion.

Reply:
Can you explain why? The method is placed right after it's (single) caller (and should actually be placed after the other methods being called from the caller, but I did not want to reorder all of them in this PR).

Reply:
Both methods do pretty much the same, except that one is working on the search tools, the other one on the RegEx tools. This is a matter of taste and rather a symptom of this class being so huge than a blocker. I will close this comment for now

Reply:
I agree that the class is quite huge and should be stripped down 🙂 
Apart from that, method placement is a rather subjective topic. Everyone has their own preference how methods should be ordered. I try to follow clean code ideas of having methods placed in a book-like reading order (not necessarily because that's a great idea but because it is some comprehensible rule to follow), but that also has its limitations.